packages/shared/lib/services/proxy.service.ts

```diff
@@ -298,6 +299,24 @@ class ProxyService {
         return false;
     };
 
+    private isRetryInHeader(retryConfig: RetryHeaderConfig | undefined, error: AxiosError): boolean {
+        if (!retryConfig) {
+            return false;
+        }
+
+        const { at, after } = retryConfig;
+
+        if (at) {
+            return Boolean(error.response?.headers[at] || error.response?.headers[at.toLowerCase()]);
```

Comment:
for later: maybe we should just enforce that headers in providers.yaml are lowercase since headers are case-insensitive

Reply:
edit: since I'm modifying the validation I'll do that